### PR TITLE
[16.0][FIX] stock_picking_delivery_link: Test compatibility with purchase

### PR DESCRIPTION
**Pull request by victoralmau**
Related to https://github.com/OCA/delivery-carrier/pull/755 (https://github.com/OCA/delivery-carrier/actions/runs/8269746616/job/22625732737)

Avoid test errors if the purchase module is installed

Please @pedrobaeza and @carolinafernandez-tecnativa can you review it?

@Tecnativa TT46563

### CODE REVIEW

**Comment by rousseldenis** [comment]
@victoralmau I don't like putting some code that depends on module that is not in dependencies...

**Comment by victoralmau** [comment]
> @victoralmau I don't like putting some code that depends on module that is not in dependencies...

Ok, I understand, any suggestions to avoid tests failing if purchase is installed?

**Comment by carolinafernandez-tecnativa** [comment]
I have checked and it looks good to me, what do you think @pedrobaeza ?

**Comment by rousseldenis** [comment]
> > @victoralmau I don't like putting some code that depends on module that is not in dependencies...
> 
> Ok, I understand, any suggestions to avoid tests failing if purchase is installed?

@victoralmau Don't check purchase module. Fill in simply the seller_ids field as this is implemented in product.

**Comment by victoralmau** [comment]
> > > @victoralmau I don't like putting some code that depends on module that is not in dependencies...
> > 
> > 
> > Ok, I understand, any suggestions to avoid tests failing if purchase is installed?
> 
> @victoralmau Don't check purchase module. Fill in simply the seller_ids field as this is implemented in product.



> > > @victoralmau I don't like putting some code that depends on module that is not in dependencies...
> > 
> > 
> > Ok, I understand, any suggestions to avoid tests failing if purchase is installed?
> 
> @victoralmau Don't check purchase module. Fill in simply the seller_ids field as this is implemented in product.

Even if we do that we will still have to confirm the purchase and the tests will fail. Please try the following use case:

- Install `stock_picking_delivery_link` + `purchase`.
- Run the tests of `stock_picking_delivery_link`

**Comment by victoralmau** [comment]
Any other proposal to solve the tests of this module? This blocks https://github.com/OCA/delivery-carrier/pull/755

**Comment by victoralmau** [comment]
Any other **better** proposals to solve this problem? This blocks several migrations.

**Comment by carolinafernandez-tecnativa** [comment]
Can we procedure to merge? So we can unblock several migrations. Thanks!

**Comment by carolinafernandez-tecnativa** [comment]
Can we move forward to merge? @pedrobaeza @rousseldenis

**Comment by carolinafernandez-tecnativa** [comment]
@pedrobaeza What do you think?

File: stock_picking_delivery_link/tests/test_delivery_link.py [hunk 1]
```diff
@@ -26,6 +26,16 @@ def setUpClass(cls):
                 "product_id": test_carrier_product.id,
             }
         )
+        # We need to know if purchase module is installed
+        cls.purchase_installed = False
```
Comment (etobella) [inline stock_picking_delivery_link/tests/test_delivery_link.py:30]: Maybe it is cleaner to use 
```suggestion
        cls.purchase_installed = cls.env.ref("base.module_purchase").state == "installed"
```

**Comment by pedrobaeza** [comment]
@rousseldenis I have been 2 hours trying to get another solution in #755, but I see no other solution than this one.

The problem is that the tests `test_put_in_pack_from_pick_with_wizard*` of `stock_picking_delivery_link` works by chance: they create the stock.move with procure_method = "make_to_order", launching a procurement that is resolved with a lucky fallback to the rule "WH: Vendors → Stock", which creates an extra move for resolving `move_orig_ids`, but that's not possible when `purchase_stock` gets installed (or at least I'm not able to force it).

Can you propose an alternative or to not block this?

**Comment by carolinafernandez-tecnativa** [comment]
Hi, @rousseldenis have you checked what @pedrobaeza commented before?

**Comment by carolinafernandez-tecnativa** [comment]
Hi, any news? we need to unlock several migrations

**Review by pedrobaeza — APPROVED** [review]
Let's merge this as the less bad solution. Later, other options can be proposed:

/ocabot merge patch

**Pull request merged**